### PR TITLE
Improve pick motion

### PR DESCRIPTION
**Pull request by YutoUchimi**
- modify overlook-pose
- modify movable region
- do not wait head motion

### CODE REVIEW

File: jsk_arc2017_baxter/euslisp/lib/arc-interface.l [hunk 1]
```diff
@@ -335,7 +336,9 @@
                   (- (/ (m->mm (send bin-box :dimensions :x)) 2.0))
                   0.0 (/ (m->mm (send bin-box :dimensions :z)) 2.0))
                 offset) :world)
-      (send bin-coords :rotate (* pi (/ coords-pitch 180)) :y)
+      (send bin-coords :rotate (deg2rad (aref rpy 1)) :y)
+      (send bin-coords :rotate (deg2rad (aref rpy 2)) :z)
+      (send bin-coords :rotate (deg2rad (aref rpy 0)) :x)
```
Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: Why do you choose this order?

Comment (knorth55) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: @pazeshun can you tell us what we should do?
Euslisp default is `xyz`?

Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: The order of `:rpy` in `make-coords` is `yaw-pitch-roll`
You can check this like below:
```
1.irteusgl$ setq a (make-cube 100 100 100)
#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0>
2.irteusgl$ setq b (make-coords)
#<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0>
3.irteusgl$ send b :scale 6
#<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0>
4.irteusgl$ objects (list a b)
;; (make-irtviewer) executed
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0>)
```
screenshot from 2017-05-20 11 23 16
```
5.irteusgl$ setq c (make-coords :rpy #f(1 0 0))
#<coordinates #X4ea4640  0.0 0.0 0.0 / 1.0 0.0 0.0>
6.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4ea4640  0.0 0.0 0.0 / 1.0 0.0 0.0>)
7.irteusgl$ send c :scale 5
#<coordinates #X4ea4640  0.0 0.0 0.0 / 1.0 0.0 0.0>
```
screenshot from 2017-05-20 11 24 53
```
8.irteusgl$ setq c (make-coords :rpy #f(0 1 0))
#<coordinates #X4f62080  0.0 0.0 0.0 / 0.0 1.0 0.0>
9.irteusgl$ send c :scale 5
#<coordinates #X4f62080  0.0 0.0 0.0 / 0.0 1.0 0.0>
10.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4f62080  0.0 0.0 0.0 / 0.0 1.0 0.0>)
```
screenshot from 2017-05-20 11 25 59
```
11.irteusgl$ setq c (make-coords :rpy #f(0 0 1))
#<coordinates #X504c798  0.0 0.0 0.0 / 0.0 0.0 1.0>
12.irteusgl$ send c :scale 5
#<coordinates #X504c798  0.0 0.0 0.0 / 0.0 0.0 1.0>
13.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X504c798  0.0 0.0 0.0 / 0.0 0.0 1.0>)
```
screenshot from 2017-05-20 11 26 33

Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: And rotating order is `roll-pitch-yaw`:
```
14.irteusgl$ setq c (make-coords :rpy #f(1 1 1))
#<coordinates #X5105708  0.0 0.0 0.0 / 1.0 1.0 1.0>
15.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X4adeb98  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X5105708  0.0 0.0 0.0 / 1.0 1.0 1.0>)
16.irteusgl$ send c :scale 5
#<coordinates #X5105708  0.0 0.0 0.0 / 1.0 1.0 1.0>
```
screenshot from 2017-05-20 11 27 34
```
31.irteusgl$ send b :rotate 1 :x :world
#<coordinates #X5242d58  0.0 0.0 0.0 / 0.0 0.0 1.0>
32.irteusgl$ send b :rotate 1 :y :world
#<coordinates #X5242d58  0.0 0.0 0.0 / 0.0 1.0 1.0>
33.irteusgl$ send b :rotate 1 :z :world
#<coordinates #X5242d58  0.0 0.0 0.0 / 1.0 1.0 1.0>
```
screenshot from 2017-05-20 11 32 12
```
34.irteusgl$ setq b (make-coords)
#<coordinates #X52128b8  0.0 0.0 0.0 / 0.0 0.0 0.0>
35.irteusgl$ send b :scale 6
#<coordinates #X52128b8  0.0 0.0 0.0 / 0.0 0.0 0.0>
36.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X52128b8  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X5105708  0.0 0.0 0.0 / 1.0 1.0 1.0>)
37.irteusgl$ send b :rotate 1 :z :world
#<coordinates #X52128b8  0.0 0.0 0.0 / 1.0 0.0 0.0>
38.irteusgl$ send b :rotate 1 :y :world
#<coordinates #X52128b8  0.0 0.0 0.0 / 1.237 0.472 0.919>
39.irteusgl$ send b :rotate 1 :x :world
#<coordinates #X52128b8  0.0 0.0 0.0 / 1.235 -0.481 1.235>
```
screenshot from 2017-05-20 11 33 25

Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: Also, rotating order in local is `yaw-pitch-roll`:
```
40.irteusgl$ setq b (make-coords)
#<coordinates #X50fe5b8  0.0 0.0 0.0 / 0.0 0.0 0.0>
41.irteusgl$ send b :scale 6
#<coordinates #X50fe5b8  0.0 0.0 0.0 / 0.0 0.0 0.0>
42.irteusgl$ objects (list a b c)
(#<body #X493f8e0 (:cube 100.0 100.0 100.0) 0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X50fe5b8  0.0 0.0 0.0 / 0.0 0.0 0.0> #<coordinates #X5105708  0.0 0.0 0.0 / 1.0 1.0 1.0>)
43.irteusgl$ send b :rotate 1 :z :local
#<coordinates #X50fe5b8  0.0 0.0 0.0 / 1.0 0.0 0.0>
44.irteusgl$ send b :rotate 1 :y :local
#<coordinates #X50fe5b8  0.0 0.0 0.0 / 1.0 1.0 0.0>
45.irteusgl$ send b :rotate 1 :x :local
#<coordinates #X50fe5b8  0.0 0.0 0.0 / 1.0 1.0 1.0>
```
screenshot from 2017-05-20 11 56 51

Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: So, we should
* change `:rpy` order to `yaw-pitch-roll`
* change `:rpy` unit to radian
* change rotating order to `yaw-pitch-roll`
ex).
```
(send bin-coords :rotate (aref rpy 0) :z :local)
(send bin-coords :rotate (aref rpy 1) :y :local)
(send bin-coords :rotate (aref rpy 2) :x :local)
```
* change `:move-arm-body->bin-overlook-pose` to follow `:rpy` change

File: jsk_arc2017_baxter/euslisp/pick-main.l [hunk 2]
```diff
@@ -94,7 +94,7 @@
           (send *ri* :angle-vector (send *baxter* :angle-vector) 3000)
           (send *ri* :wait-interpolation)
           (ros::ros-info "[main] Recognizing objects in bin ~a" target-bin)
-          (send *ri* :move-arm-body->bin-overlook-pose arm target-bin)
+          (setq avs-prev (send *ri* :move-arm-body->bin-overlook-pose arm target-bin))
```
Comment (knorth55) [inline jsk_arc2017_baxter/euslisp/pick-main.l]: We didn't use this change in Mayfes demo

File: jsk_arc2017_baxter/euslisp/pick-main.l [hunk 3]
```diff
@@ -103,7 +103,13 @@
...
             (setq recognition-trial-time (incf recognition-trial-time)))
           (if is-recognized
-            (setq state :pick_object)
+            (progn
+              (setq state :pick_object)
+              (send *ri* :angle-vector-sequence (reverse avs-prev)
+                    :fast (get-arm-controller arm) 0 :scale 5.0)
+              (send *ri* :wait-interpolation)
+              (send *ri* :move-arm-body->bin-overlook-pose arm target-bin)
+              (send *ri* :wait-interpolation))
```
Comment (knorth55) [inline jsk_arc2017_baxter/euslisp/pick-main.l]: We didn't use these changes in Mayfes demo

**Comment by knorth55** [comment]
Move to #2112

**Comment by knorth55** [comment]
#2112 is merged, so I close this PR

**Pull request closed without merging**